sdk/scheduler/src/test/java/com/mesosphere/sdk/api/JettyApiServerTest.java

```diff
@@ -75,4 +83,22 @@ public Response getFrameworkId() {
             return Response.ok(TEST, MediaType.APPLICATION_JSON).build();
         }
     }
+
+    private void sleep(int ms) {
+        // an interruptable sleep b/c if interrupted, i.e. by SIGINT, np,
+        // continue to crash, don't worry about sleep
+        try {
+            Thread.sleep(ms);
+        } catch (InterruptedException e) { }
+    }
+
+    private int getJettyApiServerPort() {
+        if (jettyApiServer == null) {
```

Comment:
Calling directly does not match the exposure and call of isStarted() . And yes, returning an error code rather than throwing an exception at the point of the exception is not correct Java. I don't think a spelunking expedition to re-find the existing code that lead me to follow it's pattern, but IMHO this is relatively unimportant as an attempt to request a resource over HTTP on port -1 or -2 will fail miserably, throw an exception and lead the developer to this line of code quickly.